x-pack/plugin/wildcard/src/test/java/org/elasticsearch/xpack/wildcard/mapper/KeywordScriptFieldApproximatedByWildcardFieldTests.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.wildcard.mapper;
+
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexableField;
+import org.apache.lucene.index.RandomIndexWriter;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanClause.Occur;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.MatchAllDocsQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+import org.apache.lucene.store.Directory;
+import org.elasticsearch.index.fielddata.ScriptDocValues;
+import org.elasticsearch.index.mapper.DocumentMapper;
+import org.elasticsearch.index.mapper.KeywordScriptFieldType;
+import org.elasticsearch.index.mapper.LuceneDocument;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.MapperServiceTestCase;
+import org.elasticsearch.index.query.SearchExecutionContext;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.queryableexpression.QueryableExpressionBuilder;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.script.ScriptType;
+import org.elasticsearch.script.StringFieldScript;
+import org.elasticsearch.script.StringFieldScript.LeafFactory;
+import org.elasticsearch.search.lookup.SearchLookup;
+import org.elasticsearch.search.runtime.AbstractScriptFieldQuery;
+import org.elasticsearch.xpack.wildcard.Wildcard;
+import org.hamcrest.Matcher;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+
+public class KeywordScriptFieldApproximatedByWildcardFieldTests extends MapperServiceTestCase {
+    public void testTermQueryApproximated() throws IOException {
```

Comment:
do you also need a test method for the `approximateSubstringQuery` functionality? e.g. that has different expectations on approximation queries for short queries like `a`

Reply:
`foo_last_word_approximated` hits it.

Reply:
Not sure I've seen the equivalent test yet in this PR but the thing that revealed most bugs for me in the wildcard field was a randomised regex generator exercising varying levels of nesting and combos of clause types and lengths. I compared unaccelerated `keyword`  field results with accelerated `wildcard` field results for equivalence and this revealed a lot of accelerator bugs.
Would it make sense to add something similar for randomising scripts and contrasting results of "optimise_first" flag on vs off?